src/portalnet/utils.rs

```diff
@@ -41,26 +42,47 @@ mod test {
 }
 
 pub fn get_data_dir() -> String {
+    let path;
     match env::var(TRIN_DATA_ENV_VAR) {
-        Ok(data_path) => data_path,
-        Err(_) => get_default_data_dir(),
+        Ok(data_path) => {
+            path = data_path;
+        }
+        Err(_) => {
+            path = get_default_data_dir();
+        }
+    }
+
+    if !Path::new(&path).exists() {
+        match fs::create_dir_all(&path) {
+            Ok(_) => {}
+            Err(_) => {
+                panic!("unable to create directory, possibly from lack of permissions");
+            }
+        }
```

Comment:
I don't think `create_dir_all` fails if the directories already exist, and using `expect` will preserve the error information:

```suggestion
    fs::create_dir_all(&path).expect("unable to create data directory")
```

Reply:
My bad, I think I got confused when reading the documentation for it.